src/agents/graph.py

```diff
@@ -0,0 +1,342 @@
+                    "- ServiceAccount"
+                    "- Role"
+                    "- RoleBinding"
+                    "- ClusterRole"
+                    "- ClusterRoleBinding"
+                    "- CustomResourceDefinition"
+                    "- Operator"
+                    "- Helm Chart"
+                    "- Taint"
+                    "- Toleration"
+                    "- Affinity"
+                    "- InitContainer"
+                    "- Sidecar"
+                    "- Kubelet"
+                    "- Kube-proxy"
+                    "- etcd"
+                    "- Kube-apiserver"
+                    "- Kube-scheduler"
+                    "- Kube-controller-manager"
+                    "- Container Runtime"
+                    "Each step/subtask should be assigned to one of these agents: {members}. "
+                    "Follow these guidelines to create the plan: "
+                    "- Carefully read and understand the given query. "
+                    "- Identify the distinct questions or tasks within the given query. "
+                    "- Use the exact wording from the original query for each list item. "
+                    "- Maintain the order of the questions as they appear in the original query. "
+                    "- Avoid too many subtasks; keep it simple and concise. "
+                    "- Avoid detailed steps; focus on the key phases or elements from the query. "
+                    "- Do not add any additional steps or explanations. ",
+                ),
+                MessagesPlaceholder(variable_name="messages"),
+            ]
+        ).partial(
+            members=", ".join(self.members),
+            output_format=self.parser.get_format_instructions(),
+        )
+        planner = planner_prompt | self.model.llm
+        try:
+            plan = planner.invoke(
+                {
+                    "messages": state.messages[-1:],
+                },  # last message is the user query
+                config={"callbacks": [handler]},
+            )
+            state.subtasks = self.parser.parse(plan.content).subtasks
+            return {
+                "subtasks": state.subtasks,
+                "messages": (
+                    [
+                        AIMessage(
+                            content=f"Task decomposed into subtasks and assigned to agents: {state.subtasks}",
+                            name=PLANNER,
+                        )
+                    ]
+                ),
+                "error": None,
+            }
+        except Exception as e:
+            logger.error(f"Error in planning: {e}")
+            return {
+                "error": str(e),
+                "messages": [
+                    AIMessage(
+                        content=f"Error in planning: {e}",
+                        name="Planner",
+                    )
+                ],
+            }
+
+    def common_node(self, state: AgentState) -> dict[str, Any]:
```

Comment:
Can you explain what is the purpose of this node?

Reply:
it answer the general questions. We'll move this to planner if possible in the next PR.